shm/Player.hpp

```diff
@@ -0,0 +1,61 @@
+#pragma once
+#include <memory>
+
+#include "Ship.hpp"
```

Comment:
When you move definition to .cpp You can make a forward declaration and include both headers in .cpp file. This causes compilation faster because any changes in both files will not trigger rebuild this one.